torch/_dynamo/variables/inline_helper.py

```diff
@@ -0,0 +1,168 @@
+from typing import Dict
+
+import torch.fx
+from .constant import ConstantVariable
+
+"""
+This file contains helper functions to decompose and inline a user function/module
+with make_fx. The primary use case of these functions are to enable the inlining of
+nn modules and torch ops during the dynamo tracing. This functionality is currently
+guarded with `torch._dynamo.config.use_single_step_graph`
+"""
+
+
+def dummy_user_function_to_inline_gm(gm, args):
+    return gm(*args)
+
+
+def dummy_user_function_to_inline_wrapped_gm(wrapped_gm, args, kwargs):
+    return wrapped_gm(args, kwargs)
+
+
+def vt_to_fake_helper(vt, tx):
+    from ..utils import get_fake_value
+
+    proxy_ = vt.as_proxy()
+
+    def proxy_to_fake_helper(p):
+        if type(p) is torch.fx.proxy.Proxy:
+            return get_fake_value(p.node, tx)
+        elif type(p) is tuple:
+            return tuple(map(proxy_to_fake_helper, p))
+        else:
+            # mostly handle scalar
+            # check return type is a fake tensor
+            assert type(p) != torch.fx.proxy.Proxy
+            return p
+
+    return proxy_to_fake_helper(proxy_)
+
+
+tracer_to_used_names: Dict[
+    torch._dynamo.output_graph.SubgraphTracer, Dict[str, int]
+] = {}
```

Comment:
actually @ezyang I can't move this dict into the tx. There are many cases that multiple different tx share the same `SubgraphTracer `. 

Another idea is for me to stick this `str` to `count` dict into the `SubgraphTracer` but it would be a bit confusing because `SubgraphTracer` uses a different method to create unique names and reader of that class might think the dict I added was for `SubgraphTracer` to create unique names.

Reply:
I don't... care if it's more confusing? Slash, this is solvable by appropriate naming. I care way more about having appropriate lifetime on this dict, and putting the used names dict on SubgraphTracer exactly sets up the ownership correctly.